dogm/demo/utils/include/metrics.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 Michael Koesel and respective contributors
+// SPDX-License-Identifier: MIT
+// See accompanying LICENSE file for detailed information
+
+#ifndef METRICS_H
+#define METRICS_H
+
+#include "precision_evaluator.h"
```

Comment:
You only depend on precision evaluator for PointWithVelocity, right? Consider extracting PointWithVelocity to a separate header (e.g. types.h). This way, you keep your modules nicely separated